feat(HTTP): Refresh 4XX response pages (#34620)
mdn/content: files/en-us/web/http/status/412/index.md

{{HTTPSidebar}}

The HTTP **`412 Precondition Failed`** status response code indicates that access to the target resource was denied.

> This happens with [conditional requests](/en-US/docs/Web/HTTP/Conditional_requests) on methods other than {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} when the condition defined by the {{HTTPHeader("If-Unmodified-Since")}} or {{HTTPHeader("If-None-Match")}} headers is not fulfilled.

Author: See https://github.com/mdn/content/issues/34965

```suggestion
This happens with [conditional requests](/en-US/docs/Web/HTTP/Conditional_requests) on methods other than {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} when the condition defined by the {{HTTPHeader("If-Unmodified-Since")}} or {{HTTPHeader("If-Match")}} headers is not fulfilled.
```

{{HTTPSidebar}}

The HTTP **`412 Precondition Failed`** status response code is part of the `400`-`499` class of [client error responses](/en-US/docs/Web/HTTP/Status#client_error_responses) and indicates that access to the target resource has been denied.

> This happens with conditional requests on methods other than {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} when the condition defined by the {{HTTPHeader("If-Unmodified-Since")}} or {{HTTPHeader("If-None-Match")}} headers is notfulfilled.

Reviewer: ```suggestion
This happens with conditional requests on methods other than {{HTTPMethod("GET")}} or {{HTTPMethod("HEAD")}} when the condition defined by the {{HTTPHeader("If-Unmodified-Since")}} or {{HTTPHeader("If-None-Match")}} headers is not fulfilled.
```

Author: nice catch, fixed in https://github.com/mdn/content/pull/34620/commits/5ec35dbdd45b72e904277ceab6f6106a745a6c10#diff-6b290415ce8c65e29a5af30900190ec6ea688d38303dd9e6ac0ca561ea9ddd7aR11

### Avoiding mid-air collisions

With the help of the `ETag` and the {{HTTPHeader("If-Match")}} headers, you can prevent conflicts or mid-air collisions.

> For example, when editing some wiki pages, content is hashed and put into an `Etag` in successful responses:

Reviewer: ```suggestion
For example, when editing some wiki pages, content is hashed and put into an `ETag` in successful responses:
```
